src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -0,0 +1,201 @@
+package org.folio.circulation.domain;
+
+import org.folio.circulation.domain.policy.OverdueFinePolicy.OverdueFineInterval;
+import org.folio.circulation.domain.policy.OverdueFinePolicyRepository;
+import org.folio.circulation.domain.representations.FeeFineRepresentation;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.ItemRepository;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ResultBinding;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.of;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ResultBinding.mapResult;
+
+public class OverdueFineCalculatorService {
+  private static class CalculationParameters {
```

Comment:
Personally I would move static classes to the bottom of the file.

Why is it preferable to have static internal classes, over putting these classes in separate files together in their own package namespace?

Reply:
Moved to the bottom. This is a helper class that won't be used anywhere outside of the OverdueFineCalculatorService. The only advantage of not moving it to a separate file I think is that amount of files in the project tree doesn't increase, which keeps navigation a bit easier.